0.download_data/download_figshare.py

```diff
@@ -0,0 +1,62 @@
+import os
+import glob
+import pathlib
+import shutil
+import urllib.request
+import zipfile
+
+
+def download_figshare(
+    figshare_id: str,
+    output_file: pathlib.Path,
+    output_dir: pathlib.Path,
+    metadata_dir: pathlib.Path,
+    figshare_url: str = "https://ndownloader.figshare.com/files/",
+    unzip_files: bool = False,
+):
+    """
+    Download the provided figshare resource and extract the files from Figshare. Extract the downloaded
+    zip file into a folder and then seperate the metadata for each plate into a seperate metadata folder.
+
+    Attributes
+    ----------
+    figshare_id : str
+        string of numbers that corresponds to the figshare identifier
+    output_file : pathlib.Path
+        the location and file name for the downloaded contents from Figshare
+    output_dir : pathlib.Path
+        path to directory for images
+    metadata_dir : pathlib.Path
+        path to directory for metadata
+    figshare_url: str, default "https://ndownloader.figshare.com/files/"
+        the location of where the figshare id is stored
+    unzip_files: bool, default False
+        if set to True, then the expected download from Figshare is a zip file which will be unzipped
+    """
+    # access the url and download the zip file from figshare containing files for plate (images + metadata)
+    urllib.request.urlretrieve(f"{figshare_url}/{figshare_id}", output_file)
+
+    if unzip_files==True:
```

Comment:
```suggestion
    if unzip_files:
```

I'm surprised `black` doesn't catch this

Reply:
I just ran it again and it changed the formatting. So if I have it defaulted False and I have `if unzip_files:`, that means the same if I had it has `== True`?

Reply:
yes, python will evaluate a boolean and enter the statement if the boolean is True

Reply:
I can confirm that this worked!